js/script.js

```diff
@@ -184,7 +185,23 @@ const markDoneTodoHandler = (inputElement, todo) => {
     renderTodos();
 };
 
-const searchHandler = (searchedValue) => {
+const debounce =(func, timeout = 500) => {
+    let timer;
+    return () => {
+        clearTimeout(timer);
+        timer = setTimeout(() => {
+            func();
+        }, timeout);
```

Comment:
Will this work if the function had arguments?

Reply:
no bhai. when we will have arguments, we have to add arguments to the function using apply or bind method. But currently, I don't have any arguments. for that I have implemented in this way.

if we had arguments, we had to implement debounce in this way. And we have to use function declaration not the function expression.

` return (...args) => {
        clearTimeout(timer);
        timer = setTimeout(() => {
            func.apply(this, args);
        }, timeout);`

Reply:
`debounce` is a generic function. Not implementing it thinking about only current project scope is a foot gun. Cause we don't know if the function is going to take arguments in the future scope or not. If it does, then it'll lead to unexpected bug.

Reply:
okay bhai I am resolving it.